test/integration/controller/jobs/deployment/deployment_webhook_test.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deployment
+
+import (
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/discovery"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	configapi "sigs.k8s.io/kueue/apis/config/v1beta1"
+	"sigs.k8s.io/kueue/pkg/controller/constants"
+	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+	"sigs.k8s.io/kueue/pkg/util/kubeversion"
+	testingdeployment "sigs.k8s.io/kueue/pkg/util/testingjobs/deployment"
+	"sigs.k8s.io/kueue/test/integration/framework"
+	"sigs.k8s.io/kueue/test/util"
+)
+
+var _ = ginkgo.Describe("Deployment Webhook", func() {
+	var ns *corev1.Namespace
+
+	ginkgo.When("with pod integration enabled", ginkgo.Ordered, ginkgo.ContinueOnFailure, func() {
+		ginkgo.BeforeAll(func() {
+			fwk = &framework.Framework{
+				CRDPath:     crdPath,
+				WebhookPath: webhookPath,
+			}
+			cfg = fwk.Init()
+
+			discoveryClient, err := discovery.NewDiscoveryClientForConfig(cfg)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			serverVersionFetcher = kubeversion.NewServerVersionFetcher(discoveryClient)
+			err = serverVersionFetcher.FetchServerVersion()
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ctx, k8sClient = fwk.RunManager(cfg, managerSetup(
+				nil,
+				jobframework.WithManageJobsWithoutQueueName(false),
+				jobframework.WithKubeServerVersion(serverVersionFetcher),
+				jobframework.WithIntegrationOptions(
+					corev1.SchemeGroupVersion.WithKind("Pod").String(),
+					&configapi.PodIntegrationOptions{},
```

Comment:
```suggestion
```

Is there any reason why we need to set empty PodIntegrationOptions?
If not, let's remove them.

Reply:
Can you just not call `WithIntegrationOptions`?

Reply:
I guess you can't because of this https://github.com/kubernetes-sigs/kueue/blob/243e572c441d0c1f305ced572151c77d6e1e9367/pkg/controller/jobs/pod/pod_webhook.go#L94

I guess we should change that function to work with an empty struct, but we can leave that for a follow up.

/lgtm
/approve

Reply:
PR was opened: https://github.com/kubernetes-sigs/kueue/pull/2945